src/system/voucher.ts

```diff
@@ -89,35 +73,34 @@ export interface Vouchers {
 export const vouchers: Vouchers = {};
 
 export function initVouchers() {
-  for (const achv of [achvs.CLASSIC_VICTORY]) {
-    const voucherType =
-      achv.score >= 150
-        ? VoucherType.GOLDEN
-        : achv.score >= 100
-          ? VoucherType.PREMIUM
-          : achv.score >= 75
-            ? VoucherType.PLUS
-            : VoucherType.REGULAR;
-    vouchers[achv.id] = new Voucher(voucherType, getAchievementDescription(achv.localizationKey));
-  }
+  const getVoucherTypeByScore = (score: number): VoucherType => {
+    if (score >= 150) return VoucherType.GOLDEN;
+    if (score >= 100) return VoucherType.PREMIUM;
+    if (score >= 75) return VoucherType.PLUS;
+    return VoucherType.REGULAR;
+  };
 
-  const bossTrainerTypes = Object.keys(trainerConfigs).filter(
-    tt =>
-      trainerConfigs[tt].isBoss &&
-      trainerConfigs[tt].getDerivedType() !== TrainerType.RIVAL &&
-      trainerConfigs[tt].hasVoucher,
+  vouchers[achvs.CLASSIC_VICTORY.id] = new Voucher(
+    getVoucherTypeByScore(achvs.CLASSIC_VICTORY.score),
+    getAchievementDescription(achvs.CLASSIC_VICTORY.localizationKey),
   );
 
-  for (const trainerType of bossTrainerTypes) {
-    const voucherType = trainerConfigs[trainerType].moneyMultiplier < 10 ? VoucherType.PLUS : VoucherType.PREMIUM;
-    const key = TrainerType[trainerType];
-    const trainerName = trainerConfigs[trainerType].name;
-    const trainer = trainerConfigs[trainerType];
-    const title = trainer.title ? ` (${trainer.title})` : "";
-    vouchers[key] = new Voucher(voucherType, `${i18next.t("voucher:defeatTrainer", { trainerName })} ${title}`);
+  for (const [key, config] of Object.entries(trainerConfigs)) {
+    if (
+      config.isBoss &&
+      config.getDerivedType() !== TrainerType.RIVAL &&
+      config.hasVoucher
+    ) {
+      const voucherType = config.moneyMultiplier < 10 ? VoucherType.PLUS : VoucherType.PREMIUM;
+      const title = config.title ? ` (${config.title})` : "";
+      vouchers[TrainerType[key as keyof typeof TrainerType]] = new Voucher(
```

Comment:
```suggestion
      vouchers[TrainerType[key]] = new Voucher(
```
This type casting doesn't appear to be necessary.